### PR TITLE
refactor: refactor pixi::consts and pixi::config into separate crates

### PR DESCRIPTION
**Pull request by tdejager**
Refactors these types into their seperate crates.

I've kept consts in the const namespace, and made the use of it consistent, i.e never import from const directly but always append with the namespace. I've kept the const module for that reason as I like to avoid `mod as other_mod` whenever possible.

Config I've also put in it seperate crate, this could use a subsequent refactor because the lib file is much too big for my taste, I think that refactor is a bit overdue tbh.

I've kept the refactor limited to moving of the types, and have not changed anything significant, I believe the review would have been impossible otherwise.

cc @wolfv

### CODE REVIEW

File: crates/pixi_consts/src/consts.rs [hunk 1]
```diff
@@ -8,9 +8,9 @@ pub use pixi_manifest::consts::*;
 pub const PROJECT_MANIFEST: &str = "pixi.toml";
```
Comment (ruben-arts) [inline crates/pixi_consts/src/consts.rs:8]: Q: Is this the best way of using a package or should these consts be moved into their respective crates. Like manifest information in `pixi_manifest` and config stuff in `pixi_config`?
Not holding back this pr, just interested in the future design.

Comment (tdejager) [reply crates/pixi_consts/src/consts.rs:8]: Yeah I tried that for this one, but ran into circular references. Should be something we think about in the future indeed.

File: crates/pixi_config/src/consts.rs
Comment (baszalmstra) [inline crates/pixi_config/src/consts.rs:1]: Is it correct that this file is empty?

File: crates/pixi_consts/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+authors.workspace = true
+edition.workspace = true
+homepage.workspace = true
+license.workspace = true
+name = "pixi_consts"
+readme.workspace = true
+repository.workspace = true
+version = "0.1.0"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
```
Comment (baszalmstra) [inline crates/pixi_consts/Cargo.toml:11]: Remove?

```suggestion
```

File: src/project/environment.rs [hunk 3]
```diff
@@ -26,7 +27,7 @@ use crate::{consts, project::has_features::HasFeatures, Project};
 /// while this struct provides methods to easily interact with an environment
 /// without having to deal with the structure of the project model.
 ///
-/// This type does not provide manipulation methods. To modify the data model
+/// This type does not provide manipulation methods. To modify the data mode
```
Comment (baszalmstra) [inline src/project/environment.rs:30]: Unintentional edit?

```suggestion
/// This type does not provide manipulation methods. To modify the data model
```

File: crates/pixi_config/src/lib.rs [hunk 4]
```diff
@@ -888,6 +896,26 @@ pub fn config_path_global() -> Vec<PathBuf> {
...
+                cache_action: Default::default(),
+            })
+            .unwrap_or_default();
+
+        Self {
+            default: default_source_config,
+            per_channel: Default::default(),
+        }
+    }
+}
```
Comment (baszalmstra) [inline crates/pixi_config/src/lib.rs:917]: Maybe move this back to pixi and use a function instead? I think this is the only reason we need the `gateway` feature of repodata_gateway or even the crate at all.

File: crates/pixi_config/src/consts.rs
Comment (tdejager) [reply crates/pixi_config/src/consts.rs:1]: Should be removed indeed.

Comment (tdejager) [reply crates/pixi_config/src/consts.rs:1]: Done.

File: crates/pixi_config/src/lib.rs (hunk 4 above)
Comment (tdejager) [reply crates/pixi_config/src/lib.rs]: Sure, done.

**Review by baszalmstra — APPROVED** [review]

**Pull request merged**